frontend/taipy-gui/src/components/Taipy/TableSort.tsx

```diff
@@ -0,0 +1,271 @@
+
+import { ColumnDesc, getsortByIndex, iconInRowSx } from "./tableUtils";
+import { getSuffixedClassNames } from "./utils";
+
+export interface SortDesc {
+    col: string;
+    order: boolean;
+}
+
+interface TableSortProps {
+    columns: Record<string, ColumnDesc>;
+    colsOrder?: Array<string>;
+    onValidate: (data: Array<SortDesc>) => void;
+    appliedSorts?: Array<SortDesc>;
+    className?: string;
+}
+
+interface SortRowProps {
+    idx: number;
+    sort?: SortDesc;
+    columns: Record<string, ColumnDesc>;
+    colsOrder: Array<string>;
+    setSort: (idx: number, fd: SortDesc, remove?: boolean) => void;
+    appliedSorts?: Array<SortDesc>;
+}
+
+const anchorOrigin = {
+    vertical: "bottom",
+    horizontal: "right",
+} as PopoverOrigin;
+
+const gridSx = { p: "0.5em", minWidth: "36rem" };
+const badgeSx = {
+    "& .MuiBadge-badge": {
+        height: "10px",
+        minWidth: "10px",
+        width: "10px",
+        borderRadius: "5px",
+    },
+};
+const orderCaptionSx = { ml: 1 };
+
+const getSortDesc = (columns: Record<string, ColumnDesc>, colId?: string, asc?: boolean) => {
+    if (colId && asc !== undefined) {
+        return {
+            col: columns[colId].dfid,
+            order: !!asc,
+        } as SortDesc;
+    }
+};
+
+const SortRow = (props: SortRowProps) => {
+    const { idx, setSort, columns, colsOrder, sort, appliedSorts } = props;
+
+    const [colId, setColId] = useState("");
+    const [order, setOrder] = useState(true); // true => asc
+    const [enableCheck, setEnableCheck] = useState(false);
+    const [enableDel, setEnableDel] = useState(false);
+
+    const cols = useMemo(() => {
+        if (!Array.isArray(appliedSorts) || appliedSorts.length == 0) {
+            return colsOrder;
+        }
+        return colsOrder.filter((col) => col == sort?.col || !appliedSorts.some((fd) => col === fd.col));
+    }, [colsOrder, appliedSorts, sort?.col]);
+
+    const onColSelect = useCallback(
+        (e: SelectChangeEvent<string>) => {
+            setColId(e.target.value);
+            setEnableCheck(!!getSortDesc(columns, e.target.value, order));
+        },
+        [columns, order]
+    );
+    const onOrderSwitch = useCallback(
+        (e: ChangeEvent<HTMLInputElement>) => {
+            setOrder(e.target.checked);
+            setEnableCheck(!!getSortDesc(columns, colId, e.target.checked));
+        },
+        [columns, colId]
+    );
+
+    const onDeleteClick = useCallback(() => setSort(idx, undefined as unknown as SortDesc, true), [idx, setSort]);
+    const onCheckClick = useCallback(() => {
+        const fd = getSortDesc(columns, colId, order);
+        fd && setSort(idx, fd);
+    }, [idx, setSort, columns, colId, order]);
+
+    useEffect(() => {
+        if (sort && idx > -1) {
+            const col = Object.keys(columns).find((col) => columns[col].dfid === sort.col) || "";
+            setColId(col);
+            setOrder(sort.order);
+            setEnableCheck(false);
```

Comment:
`false`?

Reply:
yes, it will be changed to true if something changes